iOSNativeLoginTemplate/iOSNativeLoginTemplate/NativeLoginViewFactory.swift

```diff
@@ -28,11 +28,35 @@
 import Foundation
 import SwiftUI
 
-@objc class NativeLoginViewFactory: NSObject {
-
-    @objc static func create() -> UIViewController {
-        let view = UIHostingController(rootView: NativeLoginView())
-
+class NativeLoginViewFactory: NSObject {
+    
+    static func create(
+        navigationPathObservable: NavigationPathObservable,
+        reCaptchaClientObservable: ReCaptchaClientObservable
+    ) -> UIViewController {
+        let view = NativeLoginTemplateHostingController(rootView: NativeLoginView()
+            .environmentObject(navigationPathObservable)
+            .environmentObject(reCaptchaClientObservable))
+        
         return view
     }
 }
+
+class NativeLoginTemplateHostingController<Content>: UIHostingController<Content> where Content : View {
+    
+    override func willMove(toParent parent: UIViewController?) {
+        
+        guard let navigationController = (parent as? UINavigationController) else {
+            return
+        }
+        
+        //
+        // The Salesforce Mobile SDK provided navigation controller displays a
+        // navigation bar that is redundant with the SwiftUI navigation in the
+        // template app, so hide it.
+        //
+        // TODO: See if this should be resolved in SFMSDK. ECJ20240516
```

Comment:
@brandonpage - The comment covers the issue.  Paraphrased, MSDK provides a navigation bar and SwiftUI's `NavigationStack` wants to put another _under it_.  I see lots of SwiftUI devs doing work-arounds like this where their UI is getting embedded in legacy `UINavigationController` instances.  Any ideas?  This keeps the MSDK navigation bar hidden and the template looks alright, but it's certainly not ideal.

Reply:
This sounds similar to an issue @bbirman and I were discussing about a month ago.  I don't know if a bug ever got filed for it or if Brianna make any progress into the root cause.